### PR TITLE
Fix stockwell transform for n_fft = 2**N

### PR DESCRIPTION
**Pull request by jniediek**
There were two related errors in `time_frequency/_stockwell.py`. Both were relevant only if the input data to the Stockwell transform has 2**N samples. 

1) in `_check_input_st(x_in, n_fft)`: Return was indented wrongly and zero_pad was undefined iff `n_fft == 2**N`.

2) In `_precompute_st_windows`: The computation of TFR was wrong if `zero_pad == 0`.

### CODE REVIEW

**Comment by jniediek** [comment]
Hi @dengemann, I assume you mean "tests that would have failed had the code **not** benefited from ..."? 

I can write such a test, but I need some time to familiarize myself with the test environment. I'll do.

**Comment by dengemann** [comment]
@jniediek yes, indeed *not* was missing. But you figured it out. Take a look at other PRs to find some inspiration. It's important that we cover corner cases with tests to be sure that we're not regressing when changing the code in the future.

**Comment by jniediek** [comment]
I added a test for each of the cases. They fail without the code changes and don't fail with the code changes. 

I would appreciate further comments.

File: mne/time_frequency/tests/test_stockwell.py [hunk 1]
```diff
@@ -15,13 +15,40 @@
...
+
+def test_stockwell_st_power_itc():
+    """Test stockwell power itc"""
+    data = np.zeros((20, 128))
+    start_f = 1
+    stop_f = 10
+    sfreq = 30
+    width = 2
+    W = _precompute_st_windows(data.shape[-1], start_f, stop_f, sfreq, width)
+    _st_power_itc(data, 10, True, 0, 1, W)
```
Comment (dengemann) [inline mne/time_frequency/tests/test_stockwell.py]: is this the critical test? Maybe add a comment that relates to the topic of the PR, also in the name and comment of the test.

Comment (jniediek) [inline mne/time_frequency/tests/test_stockwell.py]: The tests are 
- `test_stockwell_check_input_power_of_2`
- `test_stockwell_st_power_itc_power_of_2`

Both fail without my code changes.

File: mne/time_frequency/tests/test_stockwell.py [hunk 2]
```diff
@@ -38,8 +42,13 @@ def test_stockwell_check_input():
         assert_equal(zero_pad, 128 - last_dim)
 
 
-def test_stockwell_st_power_itc():
-    """Test stockwell power itc"""
+def test_stockwell_st_power_itc_power_of_2():
+    """
+    Test stockwell power itc
+    Handling of input data with 2**N samples per event was broken
+    Cf. https://github.com/mne-tools/mne-python/pull/3605
```
Comment (dengemann) [inline mne/time_frequency/tests/test_stockwell.py]: Ok this is a lot, have a look at the other tests in other module. Typically they have one liner comments as these get printed on testing. A good one would be:

"""Test handling of input data with 2**N samples"""

**Comment by dengemann** [comment]
You have a merge commit. This is considered as not so beautiful and sometimes can do harm; we try to avoid this. Can you do an interactive rebase and squash this merge commit? Do you need help with this?

**Comment by jniediek** [comment]
I would suggest I change the test comments one more time and open a new PR?

**Comment by dengemann** [comment]
> I would suggest I change the test comments one more time and open a new PR?

You can do this. It saves you time but also creates yet another PR. If you want to learn a bit git + gihub try to clean up this one :) We can assist you.

**Comment by jniediek** [comment]
I hope I did what you intended

**Comment by dengemann** [comment]
> I hope I did what you intended

Yes you did exactly it :) looks clean.

**Comment by jniediek** [comment]
Ok, thanks for quick replies and help! And by the way, sorry I clicked the +1 myself, it was a mistake and can only be undone with the github API, so I'll leave it.

**Comment by dengemann** [comment]
Can you checkout to master, pull, go to your branch and try a rebase?

**Comment by jniediek** [comment]
I don't succeed to create a clean PR at this stage

**Comment by dengemann** [comment]
> I don't succeed to create a clean PR at this stage

nevermind, these things can take a while to learn. How did you try to fix it?
Did you try restore the exact history of the master branch prior to your commits?
If you loose the patience just open a new PR but then from a branch freshly checked out from the current master branch head.
